src/components/SigningModals/MultiScreenModal.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Modal,
+  Viewport,
+  springs,
+  textStyle,
+  useViewport,
+  GU,
+} from '@aragon/ui'
+import { Spring, Transition, animated } from 'react-spring'
+import { useSteps } from '../../hooks'
+
+const AnimatedDiv = animated.div
+
+const DEFAULT_MODAL_WIDTH = 700
+
+function MultiScreenModal({ visible, screens, onClose }) {
+  const steps = screens.length
+  const { prev, step, next, direction } = useSteps(steps)
+
+  const [measuredHeight, setMeasuredHeight] = useState(false)
+  const [height, setHeight] = useState(null)
+  const [firstStart, setFirstStart] = useState(true)
+  const { below } = useViewport()
+
+  const smallMode = below('medium')
+  const { disableClose, width } = screens[step]
+
+  const modalWidth = width || DEFAULT_MODAL_WIDTH
+
+  const renderScreen = useCallback(
+    step => {
+      const { title, content } = screens[step]
+
+      return (
+        <React.Fragment>
+          <h1
+            css={`
+              ${smallMode ? textStyle('title4') : textStyle('title3')};
+
+              margin-top: -${1 * GU}px;
+              margin-bottom: ${2 * GU}px;
+            `}
+          >
+            {title}
+          </h1>
+          {content({
+            prevScreen: prev,
+            nextScreen: next,
+            closeModal: onClose,
+          })}
+        </React.Fragment>
+      )
+    },
+    [prev, next, onClose, screens, smallMode]
+  )
+
+  const onStart = useCallback(() => {
+    // Don’t animate or set the static height when the modal first opens
+    if (firstStart) {
+      setFirstStart(false)
+
+      return
+    }
+
+    setMeasuredHeight(true)
+  }, [firstStart])
+
+  const handleModalClose = useCallback(() => {
+    if (!disableClose) {
+      onClose()
+    }
+  }, [disableClose, onClose])
+
+  return (
+    <Viewport>
+      {({ width: viewportWidth }) => {
+        const gutter = 30
```

Comment:
It's mostly for consistency, but if this can't be a GU, let's leave a note why!

We usually also lift these variables out to the top level, named in all-caps snake case (e.g. `GUTTER_WIDTH`).